File: 229-majority-element-ii/majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        List<Integer> ans = new ArrayList<>();

        for (int ele : nums) {
            if (map.containsKey(ele)) {
                int freq = map.get(ele);
                map.put(ele, freq+1);
            } else {
                map.put(ele, 1);
            }
        }

        for (int ele : map.keySet()) {
            if (map.get(ele) > (nums.length / 3)) {
                ans.add(ele);
            }
        }
        return ans;
    }
}

File: 229-majority-element-ii/majority-element-ii.py
class Solution(object):
    def majorityElement(self, nums):
        map = {}
        for ele in nums:
            if ele in map:
                map[ele] += 1
            else:
                map[ele] = 1
        ans = []
        for key in map.keys():
            if map[key] > len(nums)/3:
                ans.append(key)
        return ans
        